### ISSUE TITLE
Listeners should be disguarded when no longer used

### ISSUE THREAD

**Issue by thoughtsunificator** [thoughtsunificator/domodel#2]
When the root element associated with a binding is removed everything that is attached to it has to be undone.

Use case: 
```javascript
import { Binding } from '/lib/domodel/index.js'

export default class extends Binding {

	async onCreated() {

		const { application } = this.props

		application.listen("some event", data => {
		}) // not cleaned up when the root element is removed

		window.addEventListener("mouseup", () => {
		}) // not cleaned up when the root element is removed

		this.root.remove() // for demo purposes but you will most likely do this inside a listener

	}

}
```

This will happen mostly when root element is removed and global event listeners or objects that are not directly related stay keeping the listener attached to them.

**Comment by thoughtsunificator** [comment]
Use a proxy.

For example in this case ``proxy.window.addEventListener("mouseup"...``

The proxy would then need to figure out how to clean references.

**Comment by thoughtsunificator** [comment]
Fixed as of bf5ddf23

````javascript
const listener = context.listen...
context.removeListener(listener)
````
